<a href="https://colab.research.google.com/github/puvanans/Financial-health-analysis/blob/main/dataPreparation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount("/content/drive")
# connect to Google Drive 

Mounted at /content/drive


In [2]:
# saving this cell for dependencies
import json

In [3]:
with open("/content/drive/My Drive/raw-data-from-API.txt","r") as file:
  data = file.read()

The above cell reads the JSON object resulting from the API call stored as 'raw-data-from-API.txt' from my Google Drive and stores it in the variable called data

In [7]:
(data)

"[{'symbol': 'TSLA', 'annualReports': [{'fiscalDateEnding': '2022-12-31', 'reportedCurrency': 'USD', 'totalAssets': '82338000000', 'totalCurrentAssets': '40917000000', 'cashAndCashEquivalentsAtCarryingValue': '16253000000', 'cashAndShortTermInvestments': '22185000000', 'inventory': '12839000000', 'currentNetReceivables': '2959000000', 'totalNonCurrentAssets': '32209000000', 'propertyPlantEquipment': '23548000000', 'accumulatedDepreciationAmortizationPPE': '9041000000', 'intangibleAssets': '409000000', 'intangibleAssetsExcludingGoodwill': '215000000', 'goodwill': '194000000', 'investments': '22185000000', 'longTermInvestments': 'None', 'shortTermInvestments': '5932000000', 'otherCurrentAssets': '2941000000', 'otherNonCurrentAssets': '4193000000', 'totalLiabilities': '36440000000', 'totalCurrentLiabilities': '26709000000', 'currentAccountsPayable': '15255000000', 'deferredRevenue': '7464000000', 'currentDebt': '1016000000', 'shortTermDebt': '530000000', 'totalNonCurrentLiabilities': '108

In [13]:
(data.replace("'",'"'))

'[{"symbol": "TSLA", "annualReports": [{"fiscalDateEnding": "2022-12-31", "reportedCurrency": "USD", "totalAssets": "82338000000", "totalCurrentAssets": "40917000000", "cashAndCashEquivalentsAtCarryingValue": "16253000000", "cashAndShortTermInvestments": "22185000000", "inventory": "12839000000", "currentNetReceivables": "2959000000", "totalNonCurrentAssets": "32209000000", "propertyPlantEquipment": "23548000000", "accumulatedDepreciationAmortizationPPE": "9041000000", "intangibleAssets": "409000000", "intangibleAssetsExcludingGoodwill": "215000000", "goodwill": "194000000", "investments": "22185000000", "longTermInvestments": "None", "shortTermInvestments": "5932000000", "otherCurrentAssets": "2941000000", "otherNonCurrentAssets": "4193000000", "totalLiabilities": "36440000000", "totalCurrentLiabilities": "26709000000", "currentAccountsPayable": "15255000000", "deferredRevenue": "7464000000", "currentDebt": "1016000000", "shortTermDebt": "530000000", "totalNonCurrentLiabilities": "108

In [24]:
raw_data = json.loads(data.replace("'",'"'))
type(raw_data)

list

The above cell serves two purpose:
*  It replaces all single-quotes with double quotes in the JSON string stored in the variable called data
* This is then converted to a python object (list) to cand stored in a variable called raw_data

In [14]:
TSLA_balance_sheets = (raw_data[0])
TSLA_stock = (raw_data[1])
TSLA_income_statement = (raw_data[2])

The above cell define three variables.

These three variables are the balance sheets, stock prices, and income statements extracted from raw_data.

Every one of these variables contains a dictionary from which we need to further extract information.

In [28]:
TSLA_quarterly_balance_sheets = TSLA_balance_sheets['quarterlyReports'] 
TSLA_quarterly_income_statement = TSLA_income_statement['quarterlyReports']
TSLA_adj_stock_price = (TSLA_stock)['Monthly Adjusted Time Series']


In [40]:
print((type(TSLA_quarterly_balance_sheets)))
print((type(TSLA_quarterly_income_statement)))
print((type(TSLA_adj_stock_price)))
print(len(TSLA_quarterly_balance_sheets))
print(len(TSLA_quarterly_income_statement))
print(len(TSLA_adj_stock_price))


<class 'list'>
<class 'list'>
<class 'dict'>
20
20
152


The above cell declares three variables:
*  'TSLA_quarterly_balance_sheets' contains a list of dictionaries(20) containing the quarterly balance sheet of the companies.
*'TSLA_quarterly_income_statements' contains a list of dictionaries(20) containing the quarterly income statements of the companies.
*'TSLA_adj_stock_price' contains a dictionary (152) containing the monthly adjusted stock price and other metrics of the company.

In [19]:
# TSLA_adj_stock_price
# TSLA_quarterly_income_statement
# TSLA_quarterly_balance_sheets

In [23]:
TSLA_quarterly_balance_sheets[0]

{'fiscalDateEnding': '2022-12-31',
 'reportedCurrency': 'USD',
 'totalAssets': '82338000000',
 'totalCurrentAssets': '40917000000',
 'cashAndCashEquivalentsAtCarryingValue': '16253000000',
 'cashAndShortTermInvestments': '22185000000',
 'inventory': '12839000000',
 'currentNetReceivables': '2959000000',
 'totalNonCurrentAssets': '32209000000',
 'propertyPlantEquipment': '23548000000',
 'accumulatedDepreciationAmortizationPPE': '9041000000',
 'intangibleAssets': '409000000',
 'intangibleAssetsExcludingGoodwill': '215000000',
 'goodwill': '194000000',
 'investments': '22185000000',
 'longTermInvestments': 'None',
 'shortTermInvestments': '5932000000',
 'otherCurrentAssets': '2941000000',
 'otherNonCurrentAssets': '4193000000',
 'totalLiabilities': '36440000000',
 'totalCurrentLiabilities': '26709000000',
 'currentAccountsPayable': '15255000000',
 'deferredRevenue': '7464000000',
 'currentDebt': '1016000000',
 'shortTermDebt': '530000000',
 'totalNonCurrentLiabilities': '10866000000',
 'c